can/interfaces/vector/canlib.py

```diff
@@ -0,0 +1,157 @@
+        LOG.debug('Open Driver: Status: %d', xl_status.value)
+        if xl_status.value == vxlapi.XL_SUCCESS:
+            channel_mask = []
+            # Get channels masks
+            for channel in self.channel:
+                app_channel = ctypes.c_uint(channel)
+                hw_type = ctypes.c_uint(0)
+                hw_index = ctypes.c_uint(0)
+                hw_channel = ctypes.c_uint(0)
+                xl_status = vxlapi.get_appl_config(
+                    app_channel=app_channel,
+                    hw_type=hw_type,
+                    hw_index=hw_index,
+                    hw_channel=hw_channel)
+                if xl_status.value == vxlapi.XL_SUCCESS:
+                    LOG.debug('Channel index %d found', app_channel.value)
+                    hw_type = ctypes.c_int(hw_type.value)
+                    hw_index = ctypes.c_int(hw_index.value)
+                    hw_channel = ctypes.c_int(hw_channel.value)
+                    channel_mask.append(
+                        vxlapi.get_channel_mask(hw_type, hw_index, hw_channel))
+                    LOG.debug('Channel %d, Type: %d, Mask: %d',
+                              hw_channel.value, hw_type.value,
+                              channel_mask[app_channel.value].value)
+            # Open one port for all channels
+            self.mask = vxlapi.XLaccess(0)
+            for CM in channel_mask:
+                self.mask.value += CM.value
+            self.port_handle = vxlapi.XLportHandle(
+                vxlapi.XL_INVALID_PORTHANDLE)
+            access_mask = self.mask
+            permission_mask = self.mask
+            xl_status = vxlapi.open_port(
+                port_handle=self.port_handle,
+                access_mask=access_mask,
+                permission_mask=permission_mask)
+            LOG.debug(
+                'Open Port: PortHandle: %d, PermissionMask: %s, Status: %d',
+                self.port_handle.value, permission_mask.value, xl_status.value)
+        return xl_status
+
+    def _go_on_bus(self):
+        xl_status = vxlapi.activate_channel(
+            port_handle=self.port_handle, access_mask=self.mask)
+        return xl_status
+
+    def recv(self, timeout=None):
+        event_count = ctypes.c_uint(1)
+        event_list = vxlapi.XLevent(0)
+        xl_status = vxlapi.receive(self.port_handle, event_count, event_list)
+        if xl_status.value != vxlapi.XL_ERR_QUEUE_IS_EMPTY:
+            if xl_status.value != vxlapi.XL_SUCCESS:
+                err_string = vxlapi.get_error_string(xl_status)
+                LOG.debug(err_string.value)
+            else:
+                ev_string = vxlapi.get_event_string(event_list)
+                if event_list.tag == vxlapi.XL_RECEIVE_MSG.value:
+                    can_msg = RX_MSG_RE(ev_string.value)
+                    if can_msg.group('id') is not None:
+                        rx_msg = Message(
+                            timestamp=long(can_msg.group('timestamp')),
```

Comment:
``timestamp`` must be in units of seconds, i.e. divide with 1000000000.0. Preferably it should also be a Unix timestamp but since the driver does not seem to support absolute timestamps it gets a bit complicated. If you want to you can look at the kvaser implementation.